c++/src/kj/common.h

```diff
@@ -1347,6 +1347,9 @@ inline T* readMaybe(T* ptr) { return ptr; }
 // "statement expressions" extension. IIFEs don't do the trick here because a lambda cannot
 // return out of the parent scope. These macros should therefore only be used in projects that
 // target GCC or GCC-compatible compilers.
+//
+// `__GNUC__` is not defined when using LLVM's MSVC-compatible compiler driver `clang-cl` (even
```

Comment:
I wonder if there are a bunch of other places where we are checking for `__GNUC__` but should also check for `__clang__`, or alternately places where we check for `_MSC_VER` for MSVC-specific work-arounds that Clang doesn't need.